handler.go

```diff
@@ -0,0 +1,63 @@
+package threebot
+
+import (
+	"context"
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/request"
+	"github.com/miekg/dns"
+)
+
+// ServeDNS implements the plugin.Handler interface.
+func (threebot *Threebot) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
+	state := request.Request{W: w, Req: r}
+	qname := state.Name()
+	qtype := state.Type()
+
+	zone := plugin.Zones(threebot.Zones).Matches(qname)
+	location := threebot.findLocation(qname, zone)
+
+	answers := make([]dns.RR, 0, 10)
+	extras := make([]dns.RR, 0, 10)
+
+	record, err := threebot.get(location)
+	if err != nil {
+		return threebot.errorResponse(state, zone, dns.RcodeBadName, nil)
+	}
+
+	switch qtype {
+	case "A":
+		answers, extras = threebot.A(qname, "", record)
+	case "AAAA":
+		answers, extras = threebot.AAAA(qname, "", record)
+	case "CNAME":
+		answers, extras = threebot.CNAME(qname, "", record)
+
+	default:
+		return threebot.errorResponse(state, zone, dns.RcodeNotImplemented, nil)
+	}
+
+
+	m := new(dns.Msg)
+	m.SetReply(r)
+	m.Authoritative = true
+	m.Answer = append(m.Answer, answers...)
+	m.Extra = append(m.Extra, extras...)
+
+	w.WriteMsg(m)
```

Comment:
WriteMsg return an error, you need to handle it.

Reply:
I checked couple of the official coredns plugins and they all seem to ignore it.